libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1719,3 +1699,190 @@ Type const& IRGeneratorForStatements::type(Expression const& _expression)
 	solAssert(_expression.annotation().type, "Type of expression not set.");
 	return *_expression.annotation().type;
 }
+
+bool IRGeneratorForStatements::visit(TryStatement const& _tryStatement)
+{
+	solAssert(!m_returnInfo.has_value(), "");
+
+	_tryStatement.externalCall().accept(*this);
+	solAssert(m_returnInfo.has_value(), "");
+	solAssert(m_returnInfo.value().functionCall == &_tryStatement.externalCall(), "");
```

Comment:
I think it would be better to use `m_returnInfo` as sparingly as possible, i.e.e it should be reset directly at this point, copied into a local variable and passed to the helper functions explicitly. What do you think?

Reply:
i like.